mpl_data_cast/gui/main.py

```diff
@@ -0,0 +1,235 @@
+        super(MPLDataCast, self).__init__(*args, **kwargs)
+
+        QtWidgets.QMainWindow.__init__(self)
+        path_ui = pkg_resources.resource_filename("mpl_data_cast.gui",
+                                                  "main.ui")
+        uic.loadUi(path_ui, self)
+
+        # settings
+        QtCore.QSettings.setDefaultFormat(QtCore.QSettings.Format.IniFormat)
+        self.settings = QtCore.QSettings()
+        # self.settings.setIniCodec("utf-8")
+
+        # load some values from the settings
+        self.widget_output.tree_depth_limit = int(self.settings.value(
+            "rtdc/tree_depth_limit", 3))
+        self.widget_input.tree_depth_limit = int(self.settings.value(
+            "rtdc/tree_depth_limit", 3))
+        self.widget_output.update_output_dir(
+            self.settings.value("rtdc/output_path", str(pathlib.Path.cwd())))
+        # signals
+        self.pushButton_transfer.clicked.connect(self.on_task_transfer)
+        # GUI
+        self.setWindowTitle(f"MPL-Data-Cast {version}")
+        # Disable native menu bar (e.g. on Mac)
+        self.menubar.setNativeMenuBar(False)
+        # File menu
+        self.actionPreferences.triggered.connect(self.on_action_preferences)
+        self.actionQuit.triggered.connect(self.on_action_quit)
+        # Help menu
+        self.actionSoftware.triggered.connect(self.on_action_software)
+        self.actionAbout.triggered.connect(self.on_action_about)
+
+        self.show()
+        self.raise_()
+
+    @QtCore.pyqtSlot()
+    def on_action_preferences(self):
+        """Show the preferences dialog"""
+        prev_tree_depth_limit = self.settings.value("rtdc/tree_depth_limit", 8)
+        dlg = preferences.Preferences(self)
+        dlg.setWindowTitle("MPL-Data-Cast Preferences")
+        dlg.exec()
+        # update maximum tree depth if necessary
+        if self.settings.value("rtdc/tree_depth_limit",
+                               8) != prev_tree_depth_limit:
+            self.widget_output.tree_depth_limit = int(self.settings.value(
+                "rtdc/tree_depth_limit", 8))
+            self.widget_input.tree_depth_limit = int(self.settings.value(
+                "rtdc/tree_depth_limit", 8))
+            if self.widget_output.path is not None:
+                self.widget_output.update_output_dir(self.widget_output.path)
+            if self.widget_input.path is not None:
+                self.widget_input.update_input_dir(self.widget_input.path)
+
+    @QtCore.pyqtSlot()
+    def on_action_quit(self) -> None:
+        """Determine what happens when the user wants to quit"""
+        QtCore.QCoreApplication.quit()
+
+    def on_action_about(self) -> None:
+        """Show imprint."""
+        gh = "GuckLab/MPL-Data-Cast"
+        rtd = "mpl-data-cast.readthedocs.io"
+        about_text = "Convert and transfer data from measurement PCs to " \
+                     "network shares at MPL.<br><br>" \
```

Comment:
You can remove the MPL part here, because the program should work on any infrastructure.

Reply:
Good point!